src/routing_table/network_tests.rs

```diff
@@ -109,6 +120,34 @@ impl Network {
                 }
             }
         }
+
+        let mut merge_other_info = Vec::new();
+
+        // handle broadcast of merge_own_group
+        for (targets, merge_own_details) in merge_own_info {
+            for target in targets {
+                let target_node = unwrap!(self.nodes.get_mut(&target));
+                let other_info = target_node.merge_own_group(&merge_own_details);
+                merge_other_info.push(other_info);
+                // add needed contacts
+                let needed = target_node.needed().clone();
+                for needed_contact in needed {
+                    let _ = target_node.add(needed_contact);
```

Comment:
`target_node` should probably also be added to `needed_contact`'s routing table.